### ISSUE TITLE
Extention throws error when multiple instances of VS Code is opened via CLI (CMD/PowerShell)

### ISSUE THREAD

**Issue by PairedPrototype** [streetsidesoftware/vscode-spell-checker#709]
When you open a second instance of VS Code using `code .`  on the CLI, the Code Spell Checker extension throws an error in the log output and also causes a `debug.log` file to be created in the root of the workspace which is locked until that second instance of VS Code is closed.

It's pretty easily reproducible:
(Tested only on Windows)

1. Open VS Code.
2. Open CMD/PowerShell
3. cd to a directory of your choosing
4. type `code .`
5. Observe a new `debug.log` file created at the root of the second instance's workspace
6. Observe an error in the Output tab of the bottom panel in the second instance with "Code Spell Checker" selected in the dropdown 

Example of the error in the logs and in the `debug.log` file that gets created.
```
[0218/003833.586:ERROR:registration_protocol_win.cc(102)] CreateFile: The system cannot find the file specified. (0x2)
```
This doesn't seem to be an issue if you open a second instance using the Windows start menu or desktop icon.

Code Spell Checker does, however, seem to work still in the second instance, so this issue is probably quite a low impact/importance.

**Comment by Jason3S** [comment]
Thank you. I would have never seen this.

I don't have a Windows setup easily available.

Would you set the spell checker's log level to debug: `Settings` -> `cspell log level`
![image](https://user-images.githubusercontent.com/3740137/108317103-18478f80-71be-11eb-9075-c4dce84c3215.png)

I'm interested in the lines around the error you are seeing.

The output would be something like this:
```
1	2021-02-18T06:47:44.485Z	Starting Spell Checker Server	
2	2021-02-18T06:47:44.599Z	Create Connection	
3	2021-02-18T06:47:44.608Z	onInitialize	
4	2021-02-18T06:47:44.608Z	setWorkspaceBase URI: file:///Users/jason/projects/cspell5/packages/cspell-lib/samples	
[Info  - 7:47:44 a.m.] 5	2021-02-18T06:47:44.714Z	Configuration Change	
6	2021-02-18T06:47:44.714Z	Trigger Update Config	
7	2021-02-18T06:47:44.715Z	registerConfigurationFile:	/Users/jason/.vscode/extensions/streetsidesoftware.code-spell-checker-czech-0.1.6/node_modules/cspell-dict-cs-cz/cspell-ext.json
8	2021-02-18T06:47:44.715Z	resetSettings	
[Info  - 7:47:44 a.m.] 9	2021-02-18T06:47:44.715Z	Register Configuration File	/Users/jason/.vscode/extensions/streetsidesoftware.code-spell-checker-czech-0.1.6/node_modules/cspell-dict-cs-cz/cspell-ext.json
10	2021-02-18T06:47:44.715Z	Trigger Update Config	
11	2021-02-18T06:47:44.716Z	registerConfigurationFile:	/Users/jason/.vscode/extensions/streetsidesoftware.code-spell-checker-dutch-0.2.6/node_modules/cspell-dict-nl-nl/cspell-ext.json
12	2021-02-18T06:47:44.716Z	resetSettings	
[Info  - 7:47:44 a.m.] 13	2021-02-18T06:47:44.716Z	Register Configuration File	/Users/jason/.vscode/extensions/streetsidesoftware.code-spell-checker-dutch-0.2.6/node_modules/cspell-dict-nl-nl/cspell-ext.json
14	2021-02-18T06:47:44.716Z	Trigger Update Config	
```

**Comment by PairedPrototype** [comment]
Sure, can do. Here's the output I receive: 

```
[0218/090400.196:ERROR:registration_protocol_win.cc(102)] CreateFile: The system cannot find the file specified. (0x2)
1	2021-02-18T09:04:01.013Z	Starting Spell Checker Server	
2	2021-02-18T09:04:01.204Z	Create Connection	
3	2021-02-18T09:04:01.214Z	onInitialize	
4	2021-02-18T09:04:01.214Z	setWorkspaceBase URI: file:///c%3A/Users/PairedPrototype/source/repos/infastructure/dns	
[Info  - 09:04:01] 5	2021-02-18T09:04:01.220Z	Configuration Change	
6	2021-02-18T09:04:01.220Z	Trigger Update Config	
7	2021-02-18T09:04:01.226Z	Request Validate: v1	.../dnsconfig.js
8	2021-02-18T09:04:01.226Z	getUriSettings:	.../dnsconfig.js
9	2021-02-18T09:04:01.226Z	Start fetchUriSettings:	.../dnsconfig.js
10	2021-02-18T09:04:01.227Z	Start fetchUriSettingsEx:	.../dnsconfig.js
11	2021-02-18T09:04:01.227Z	fetchFolderSettings: URI file:///c%3A/Users/PairedPrototype/source/repos/infastructure/dns/dnsconfig.js	
12	2021-02-18T09:04:01.227Z	getConfiguration	.../dnsconfig.js
13	2021-02-18T09:04:01.236Z	setWorkspaceFolders folders URI: [file:///c%3A/Users/PairedPrototype/source/repos/infastructure/dns]	
14	2021-02-18T09:04:01.236Z	setWorkspaceBase URI: file:///c%3A/Users/PairedPrototype/source/repos/infastructure/dns	
15	2021-02-18T09:04:01.245Z	readSettingsFiles:	
16	2021-02-18T09:04:01.245Z	resolveConfigImports:	...
17	2021-02-18T09:04:01.245Z	resolvingConfigImports: [
]	
18	2021-02-18T09:04:01.245Z	resolvingConfigImports ABS: [
]	
19	2021-02-18T09:04:01.251Z	importSettings	
20	2021-02-18T09:04:01.251Z	readSettingsFiles:	
21	2021-02-18T09:04:01.251Z	Finish fetchUriSettingsEx:	.../dnsconfig.js
22	2021-02-18T09:04:01.254Z	Finish fetchUriSettings:	.../dnsconfig.js
[Info  - 09:04:01] 23	2021-02-18T09:04:01.311Z	Validate File	.../dnsconfig.js
24	2021-02-18T09:04:01.311Z	validateTextDocument start: v1	.../dnsconfig.js
25	2021-02-18T09:04:01.351Z	updateActiveSettings	
26	2021-02-18T09:04:01.351Z	resetSettings	
27	2021-02-18T09:04:01.864Z	Validate all documents	
28	2021-02-18T09:04:01.864Z	Request Validate: v1	.../dnsconfig.js
29	2021-02-18T09:04:01.864Z	getUriSettings:	.../dnsconfig.js
30	2021-02-18T09:04:01.864Z	Start fetchUriSettings:	.../dnsconfig.js
31	2021-02-18T09:04:01.864Z	Start fetchUriSettingsEx:	.../dnsconfig.js
32	2021-02-18T09:04:01.864Z	fetchFolderSettings: URI file:///c%3A/Users/PairedPrototype/source/repos/infastructure/dns/dnsconfig.js	
33	2021-02-18T09:04:01.864Z	getConfiguration	.../dnsconfig.js
34	2021-02-18T09:04:01.866Z	Request Validate: v1	.../Program%20Files/Microsoft%20VS%20Code/resources/app/extensions/node_modules/typescript/lib/lib.es5.d.ts
35	2021-02-18T09:04:01.866Z	getUriSettings:	.../Program%20Files/Microsoft%20VS%20Code/resources/app/extensions/node_modules/typescript/lib/lib.es5.d.ts
36	2021-02-18T09:04:01.866Z	Start fetchUriSettings:	.../Program%20Files/Microsoft%20VS%20Code/resources/app/extensions/node_modules/typescript/lib/lib.es5.d.ts
37	2021-02-18T09:04:01.866Z	Start fetchUriSettingsEx:	.../Program%20Files/Microsoft%20VS%20Code/resources/app/extensions/node_modules/typescript/lib/lib.es5.d.ts
38	2021-02-18T09:04:01.866Z	fetchFolderSettings: URI file:///c%3A/Program%20Files/Microsoft%20VS%20Code/resources/app/extensions/node_modules/typescript/lib/lib.es5.d.ts	
39	2021-02-18T09:04:01.866Z	getConfiguration	.../Program%20Files/Microsoft%20VS%20Code/resources/app/extensions/node_modules/typescript/lib/lib.es5.d.ts
40	2021-02-18T09:04:02.272Z	validateTextDocument done: v1	.../dnsconfig.js
41	2021-02-18T09:04:02.278Z	Request Validate: v1	.../Program%20Files/Microsoft%20VS%20Code/resources/app/extensions/node_modules/typescript/lib/lib.dom.d.ts
42	2021-02-18T09:04:02.278Z	getUriSettings:	.../Program%20Files/Microsoft%20VS%20Code/resources/app/extensions/node_modules/typescript/lib/lib.dom.d.ts
43	2021-02-18T09:04:02.278Z	Start fetchUriSettings:	.../Program%20Files/Microsoft%20VS%20Code/resources/app/extensions/node_modules/typescript/lib/lib.dom.d.ts
44	2021-02-18T09:04:02.278Z	Start fetchUriSettingsEx:	.../Program%20Files/Microsoft%20VS%20Code/resources/app/extensions/node_modules/typescript/lib/lib.dom.d.ts
45	2021-02-18T09:04:02.278Z	fetchFolderSettings: URI file:///c%3A/Program%20Files/Microsoft%20VS%20Code/resources/app/extensions/node_modules/typescript/lib/lib.dom.d.ts	
46	2021-02-18T09:04:02.278Z	getConfiguration	.../Program%20Files/Microsoft%20VS%20Code/resources/app/extensions/node_modules/typescript/lib/lib.dom.d.ts
47	2021-02-18T09:04:02.292Z	readSettingsFiles:	
48	2021-02-18T09:04:02.292Z	resolveConfigImports:	...
49	2021-02-18T09:04:02.292Z	resolvingConfigImports: [
]	
50	2021-02-18T09:04:02.292Z	resolvingConfigImports ABS: [
]	
51	2021-02-18T09:04:02.292Z	resolveConfigImports:	.../
52	2021-02-18T09:04:02.292Z	resolvingConfigImports: [
]	
53	2021-02-18T09:04:02.292Z	resolvingConfigImports ABS: [
]	
54	2021-02-18T09:04:02.293Z	readSettingsFiles:	
55	2021-02-18T09:04:02.294Z	resolveConfigImports:	.../
56	2021-02-18T09:04:02.294Z	resolvingConfigImports: [
]	
57	2021-02-18T09:04:02.294Z	resolvingConfigImports ABS: [
]	
58	2021-02-18T09:04:02.294Z	importSettings	
59	2021-02-18T09:04:02.294Z	readSettingsFiles:	
60	2021-02-18T09:04:02.294Z	Finish fetchUriSettingsEx:	.../dnsconfig.js
61	2021-02-18T09:04:02.294Z	Finish fetchUriSettingsEx:	.../Program%20Files/Microsoft%20VS%20Code/resources/app/extensions/node_modules/typescript/lib/lib.es5.d.ts
62	2021-02-18T09:04:02.294Z	Finish fetchUriSettingsEx:	.../Program%20Files/Microsoft%20VS%20Code/resources/app/extensions/node_modules/typescript/lib/lib.dom.d.ts
63	2021-02-18T09:04:02.294Z	Finish fetchUriSettings:	.../dnsconfig.js
64	2021-02-18T09:04:02.294Z	Finish fetchUriSettings:	.../Program%20Files/Microsoft%20VS%20Code/resources/app/extensions/node_modules/typescript/lib/lib.es5.d.ts
65	2021-02-18T09:04:02.294Z	Finish fetchUriSettings:	.../Program%20Files/Microsoft%20VS%20Code/resources/app/extensions/node_modules/typescript/lib/lib.dom.d.ts
[Info  - 09:04:02] 66	2021-02-18T09:04:02.365Z	Validate File	.../dnsconfig.js
67	2021-02-18T09:04:02.365Z	validateTextDocument start: v1	.../dnsconfig.js
68	2021-02-18T09:04:02.376Z	validateTextDocument done: v1	.../dnsconfig.js
69	2021-02-18T09:04:02.376Z	validateTextDocument skip:	.../Program%20Files/Microsoft%20VS%20Code/resources/app/extensions/node_modules/typescript/lib/lib.es5.d.ts
70	2021-02-18T09:04:02.376Z	validateTextDocument skip:	.../Program%20Files/Microsoft%20VS%20Code/resources/app/extensions/node_modules/typescript/lib/lib.dom.d.ts
71	2021-02-18T09:04:03.638Z	Request Validate: v1	.../Program%20Files/Microsoft%20VS%20Code/resources/app/extensions/node_modules/typescript/lib/lib.webworker.importscripts.d.ts
72	2021-02-18T09:04:03.638Z	getUriSettings:	.../Program%20Files/Microsoft%20VS%20Code/resources/app/extensions/node_modules/typescript/lib/lib.webworker.importscripts.d.ts
73	2021-02-18T09:04:03.638Z	Start fetchUriSettings:	.../Program%20Files/Microsoft%20VS%20Code/resources/app/extensions/node_modules/typescript/lib/lib.webworker.importscripts.d.ts
74	2021-02-18T09:04:03.638Z	Start fetchUriSettingsEx:	.../Program%20Files/Microsoft%20VS%20Code/resources/app/extensions/node_modules/typescript/lib/lib.webworker.importscripts.d.ts
75	2021-02-18T09:04:03.638Z	fetchFolderSettings: URI file:///c%3A/Program%20Files/Microsoft%20VS%20Code/resources/app/extensions/node_modules/typescript/lib/lib.webworker.importscripts.d.ts	
76	2021-02-18T09:04:03.638Z	getConfiguration	.../Program%20Files/Microsoft%20VS%20Code/resources/app/extensions/node_modules/typescript/lib/lib.webworker.importscripts.d.ts
77	2021-02-18T09:04:03.642Z	resolveConfigImports:	.../
78	2021-02-18T09:04:03.642Z	resolvingConfigImports: [
]	
79	2021-02-18T09:04:03.642Z	resolvingConfigImports ABS: [
]	
80	2021-02-18T09:04:03.643Z	Finish fetchUriSettingsEx:	.../Program%20Files/Microsoft%20VS%20Code/resources/app/extensions/node_modules/typescript/lib/lib.webworker.importscripts.d.ts
81	2021-02-18T09:04:03.643Z	Finish fetchUriSettings:	.../Program%20Files/Microsoft%20VS%20Code/resources/app/extensions/node_modules/typescript/lib/lib.webworker.importscripts.d.ts
82	2021-02-18T09:04:03.646Z	Request Validate: v1	.../Program%20Files/Microsoft%20VS%20Code/resources/app/extensions/node_modules/typescript/lib/lib.scripthost.d.ts
83	2021-02-18T09:04:03.646Z	getUriSettings:	.../Program%20Files/Microsoft%20VS%20Code/resources/app/extensions/node_modules/typescript/lib/lib.scripthost.d.ts
84	2021-02-18T09:04:03.646Z	Start fetchUriSettings:	.../Program%20Files/Microsoft%20VS%20Code/resources/app/extensions/node_modules/typescript/lib/lib.scripthost.d.ts
85	2021-02-18T09:04:03.646Z	Start fetchUriSettingsEx:	.../Program%20Files/Microsoft%20VS%20Code/resources/app/extensions/node_modules/typescript/lib/lib.scripthost.d.ts
86	2021-02-18T09:04:03.646Z	fetchFolderSettings: URI file:///c%3A/Program%20Files/Microsoft%20VS%20Code/resources/app/extensions/node_modules/typescript/lib/lib.scripthost.d.ts	
87	2021-02-18T09:04:03.646Z	getConfiguration	.../Program%20Files/Microsoft%20VS%20Code/resources/app/extensions/node_modules/typescript/lib/lib.scripthost.d.ts
88	2021-02-18T09:04:03.649Z	resolveConfigImports:	.../
89	2021-02-18T09:04:03.649Z	resolvingConfigImports: [
]	
90	2021-02-18T09:04:03.649Z	resolvingConfigImports ABS: [
]	
91	2021-02-18T09:04:03.650Z	Finish fetchUriSettingsEx:	.../Program%20Files/Microsoft%20VS%20Code/resources/app/extensions/node_modules/typescript/lib/lib.scripthost.d.ts
92	2021-02-18T09:04:03.650Z	Finish fetchUriSettings:	.../Program%20Files/Microsoft%20VS%20Code/resources/app/extensions/node_modules/typescript/lib/lib.scripthost.d.ts
93	2021-02-18T09:04:03.697Z	validateTextDocument skip:	.../Program%20Files/Microsoft%20VS%20Code/resources/app/extensions/node_modules/typescript/lib/lib.webworker.importscripts.d.ts
94	2021-02-18T09:04:03.712Z	validateTextDocument skip:	.../Program%20Files/Microsoft%20VS%20Code/resources/app/extensions/node_modules/typescript/lib/lib.scripthost.d.ts

```

**Comment by Jason3S** [comment]
It is happening before the extension server has started.

It seems to be a VS Code issue.  See: [[1210/111511.666:ERROR:registration_protocol_win.cc(103)] CreateFile: The system cannot find the file specified. (0x2) · Issue #112216 · microsoft/vscode](https://github.com/microsoft/vscode/issues/112216)

**Comment by Jason3S** [comment]
I'll close this for now, since it seems to be a VS Code issue.
Thank you for taking the time to report it.

**Comment by PairedPrototype** [comment]
I did think that as well since I Googled the error initially, but it only seems to occur when this extension is enabled. I tried `code --extensions-disabled .` and I get no error or `debug.log` file created.

I can also just disable the CSpell extension and then do a normal `code .` and it will still launch without errors and without the file being created.

As soon as CSpell is re-enabled, the issue returns.

**Comment by Jason3S** [comment]
I think it is any extension that has a language server will cause this to happen.

See: [v1.49 creates debug.log automatically · Issue #105191 · microsoft/vscode](https://github.com/microsoft/vscode/issues/105191#issuecomment-695349628)


This extension uses

**Comment by PairedPrototype** [comment]
Ah see, I guess they've had a regression then since it was supposedly fixed in `1.49.2` and I'm on `1.53.0` 🤔

It's not really harmful, just annoying 😁 I can live with that until a fix is sorted out 👍

Thanks for the help though. 🙂

**Comment by Jason3S** [comment]
I see a code fix was merged a few days ago.
